GVRf/Framework/framework/src/main/java/org/gearvrf/GVRRenderTarget.java

```diff
@@ -0,0 +1,134 @@
+ * GVRRenderTarget can initiate a DrawFrameListener which causes
+ * the scene to be rendered into the texture every frame.
+ * To initiate rendering you must call @{link #startListening()}.
+ * A render target may also have a custom camera to allow control
+ * over the projection matrix.
+ * @see GVRRenderTexture
+ * @see GVRShadowMap
+ */
+public class GVRRenderTarget extends GVRBehavior
+{
+    protected GVRCamera mCamera;
+    protected GVRMaterial mMaterial;
+    protected GVRRenderTexture mTexture;
+    protected GVRScene  mScene;
+
+    static public long getComponentType()
+    {
+        return NativeRenderTarget.getComponentType();
+    }
+
+    /**
+     * Constructs a render target component which renders the given scene to a designated texture.
+     * The objects in the scene are rendered from the viewpoint of the scene object
+     * the GVRRenderTarget is attached to. Nothing is rendered if
+     * the render target is not attached to a scene object.
+     * You must call @{link #setEnable(true)} to initiate rendering.
+     *
+     * @param texture   GVRRenderTexture to render to.
+     * @param scene     GVRScene to render.
+     * @see #setEnable(boolean)
+     */
+    public GVRRenderTarget(GVRRenderTexture texture, GVRScene scene)
+    {
+        super(texture.getGVRContext(), NativeRenderTarget.ctor(texture.getNative()));
+        setEnable(false);
+        mTexture = texture;
+        mScene = scene;
+        setCamera(scene.getMainCameraRig().getCenterCamera());
+    }
+
+    /**
+     * Internal constructor for subclasses.
+     * @param ctx
+     * @param nativePointer
+     */
+    protected GVRRenderTarget(GVRContext ctx, long nativePointer)
+    {
+        super(ctx, nativePointer);
+    }
+
+    /**
+     * Sets the camera to use for this render target.
+     * The position and orientation of the render camera is taken
+     * from the transform of the scene object this component is
+     * attached to. The projection matrix which defines the
+     * camera view volume is taken from this camera.
+     * If no camera is provided, the projection matrix from
+     * the scene's main camera is used.
+     * @param camera GVRCamera to use for rendering this render target.
+     * @see #getCamera()
+     */
+    public void setCamera(GVRCamera camera)
+    {
+        mCamera = camera;
+        NativeRenderTarget.setCamera(getNative(), camera.getNative());
+    }
+
+    /**
+     * Gets the camera used to render to this render target.
+     * @return GVRCamera used for rendering
+     * @see #setCamera(GVRCamera)
+     */
+    public GVRCamera getCamera()
+    {
+        return mCamera;
+    }
+
+    /**
+     * Sets the texture this render target will render to.
+     * If no texture is provided, the render target will
+     * not render anything.
+     * @param texture GVRRenderTexture to render to.
+     */
+    public void setTexture(GVRRenderTexture texture)
```

Comment:
Might be better if it is package private. Having this component should imply creating the render texture for the user. getTexture should probably return just a GVRTexture. Ideally end users should not be aware of GVRRenderTexture.